### DISCUSSION TITLE
Manually setting `isDirty = true | false`

### DISCUSSION THREAD

**Discussion by bstro** [react-hook-form/react-hook-form#3913] [Q&A]
Is there any reason why one ought not to set `isDirty` by hand? Is this a derived value that react-hook-form calculates on its own, such that it would be dangerous to set it manually, I.e. `formState.isDirty = true` ?

Thanks!

**Comment by bluebill1049** [comment]
you can use `setValue('xx', 'xxx' { shouldDirty: true })`.

**Reply by diliprai-omiba** [reply]
Can we set manually dirty fields to false

**Reply by Moshyfawn** [reply]
No

**Reply by Ganbin** [reply]
You can if after setValue the value is the same as the default, you can do a trigger().
As I understand isDirty means the value is different from the default. 
It would be great to have a shouldValidate property to the resetField function.
